### PR TITLE
Add CrudVisibility component for implementing visibility column in a Crud Grid

### PR DESCRIPTION
**Pull request by nsams**
This should be best practice and will be used by admin generator

### CODE REVIEW

File: packages/api/cms-api/src/generator/generate-crud-input.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export async function generateCrudInput(generatorOptions: { targetDirectory: str
         } else {
             decorators.push("@IsOptional()");
         }
-        if (prop.name === "id" || prop.name == "createdAt" || prop.name == "updatedAt") {
+        if (prop.name === "id" || prop.name == "createdAt" || prop.name == "updatedAt" || prop.name == "visible") {
```
Comment (dkarnutsch) [inline packages/api/cms-api/src/generator/generate-crud-input.ts:36]: OPT

```suggestion
        if (["id", "createdAt", "updatedAt", "visible"].includes(prop.name)) {
```

File: packages/admin/admin/src/dataGrid/CrudVisibility.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        handleMenuClose();
+        snackbarApi.showSnackbar(
+            <UndoSnackbar
+                message={
+                    <FormattedMessage
+                        id="comet.common.visibilityChanged"
+                        defaultMessage="Changed visibility to {visibility, select, visible {published} hidden {unpublished} other {unknown}}"
+                        values={{
+                            visibility: visibility ? "visible" : "hidden",
+                        }}
```
Comment (johnnyomair) [inline packages/admin/admin/src/dataGrid/CrudVisibility.tsx:37]: Shouldn't using the boolean work directly?

```tsx
<FormattedMessage
  id="comet.common.visibilityChanged"
  defaultMessage="Changed visibility to {visibility, select, true {published} false {unpublished} other {unknown}}"
  values={{ visibility }}
/>
```

File: packages/admin/admin/src/dataGrid/CrudVisibilityIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Disabled, Online } from "@comet/admin-icons";
+import { useTheme } from "@mui/material";
+import React from "react";
+
+interface CrudVisibilityIconProps {
+    visibility: boolean;
+    disabled?: boolean;
+}
+
+export function CrudVisibilityIcon({ visibility, disabled }: CrudVisibilityIconProps): JSX.Element {
```
Comment (johnnyomair) [inline packages/admin/admin/src/dataGrid/CrudVisibilityIcon.tsx:10]:
```suggestion
export function CrudVisibilityIcon({ visibility }: CrudVisibilityIconProps): JSX.Element {
```

File: packages/admin/admin/src/dataGrid/CrudVisibilityIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Disabled, Online } from "@comet/admin-icons";
+import { useTheme } from "@mui/material";
+import React from "react";
+
+interface CrudVisibilityIconProps {
+    visibility: boolean;
+    disabled?: boolean;
```
Comment (johnnyomair) [inline packages/admin/admin/src/dataGrid/CrudVisibilityIcon.tsx:7]:
```suggestion
```

**Review by johnnyomair — CHANGES_REQUESTED** [review]

**Review by johnnyomair — DISMISSED** [review]

**Review by johnnyomair — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**